### PR TITLE
Kernel: Fix a number of problems with the software context switching

### PR DESCRIPTION
**Pull request by tomuta**
I believe this should fix the hanging processes issue as reported in #2693

### CODE REVIEW

File: Kernel/Heap/SlabAllocator.cpp [hunk 1]
```diff
@@ -130,6 +131,7 @@ void slab_alloc_init()
     s_slab_allocator_16.init(128 * KB);
     s_slab_allocator_32.init(128 * KB);
     s_slab_allocator_64.init(512 * KB);
+    s_slab_allocator_64.init(512 * KB);
```
Comment (awesomekling) [inline Kernel/Heap/SlabAllocator.cpp:134]: Not initializing the right bucket here ;)

Comment (tomuta) [reply Kernel/Heap/SlabAllocator.cpp:134]: Whoops, not sure what bad things this caused...

**Comment by tomuta** [comment]
Ok cleaned up a few more cosmetic things and log statements, I think this is ready to be merged

**Comment by nico** [comment]
I patched this in and things seem to work again on my system with this applied.

**Comment by tomuta** [comment]
It still may hang early in the boot process every now and then with -smp 2 or more, I have another branch that I started working on before this issue that I think will help address this (basically inter-cpu messaging/interrupts for MM etc). It'll be a challenge rebasing that on top of these changes, but maybe for now we should change run.sh to use -smp 1 by default

**Comment by awesomekling** [comment]
Stellar, thank you. :)

**Pull request merged**